src/components/input/Select.tsx

```diff
@@ -112,15 +124,29 @@ function SelectOption<T>({
         },
         classes,
       )}
-      onClick={() => {
-        if (!disabled) {
-          selectOrToggle();
+      onClick={e => {
+        // Do not invoke callback if clicked element is the checkbox, as it has
+        // its own event handler.
+        if (!disabled && e.target !== checkboxRef.current) {
+          selectOneValue();
         }
       }}
-      onKeyPress={e => {
-        if (!disabled && ['Enter', 'Space'].includes(e.code)) {
+      onKeyDown={e => {
```

Comment:
This should have always been `onKeyDown`, not `onKeyPress`, as the `keypress` event [is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event).